### PR TITLE
Adding documentation for Kanister Function KubeOps & Wait 

### PR DESCRIPTION
**Pull request by shlokchaudhari9**
## Change Overview

Adding the documentation for Kanister Function KubeOps & Wait

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ✅ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #1130 

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: docs/functions.rst [hunk 1]
```diff
@@ -1166,6 +1166,60 @@ Example:
       args:
         snapshotID: "{{ .ArtifactsIn.backupInfo.KeyValue.snapshotID }}"
 
+
+KubeOps
+-------
+
+This function is used to manage Kubernetes resources. We can use it to
```
Comment (viveksinghggits) [inline docs/functions.rst:1173]: instead of manage can we just use create/delete in the first statement itself?

File: docs/functions.rst [hunk 2]
```diff
@@ -1166,6 +1166,60 @@ Example:
...
+either create or delete resources on Kubernetes cluster.
+
+Arguments:
+
+.. csv-table::
+   :header: "Argument", "Required", "Type", "Description"
+   :align: left
+   :widths: 5,5,5,15
+
+   `operation`, Yes, `string`, ``create`` or ``delete`` kubernetes resource
```
Comment (viveksinghggits) [inline docs/functions.rst:1183]:
```suggestion
   `operation`, Yes, `string`, ``create`` or ``delete`` Kubernetes resource
```

File: docs/functions.rst (hunk 1 above)
Comment (shlokchaudhari9) [reply docs/functions.rst:1173]: I mentioned manage because more operations will be added to this function

Comment (shlokchaudhari9) [reply docs/functions.rst:1173]: If mentioning create/delete specifically sounds more correct then I'll make this change.

File: docs/functions.rst [hunk 3]
```diff
@@ -1166,6 +1166,99 @@ Example:
...
+
+   `timeout`, Yes, `string`, wait timeout
+   `conditions`, Yes, `map[string]interface{}`, list of wait conditions that match ``anyOf`` or ``allOf`` the conditions
+
+Example:
+
+.. code-block:: yaml
+  :linenos:
+
+  - func: Wait
```
Comment (pavannd1) [inline docs/functions.rst:1244]: Something isn't right with this block
```
Warning, treated as error:
/repo/docs/functions.rst:1241:Could not lex literal_block as "yaml". Highlighting skipped.
```

File: docs/functions.rst [hunk 4]
```diff
@@ -1166,6 +1166,99 @@ Example:
...
+                - containerPort: 80
+                  name: http
+                  protocol: TCP
+
+
+Wait
+----
+
+This function is used to add wait on a Kubernetes resource
+till it matches a desired state.
```
Comment (pavannd1) [inline docs/functions.rst:1227]:
```suggestion
This function is used to wait on a Kubernetes resource
until a desired state is reached.
```

File: docs/functions.rst [hunk 5]
```diff
@@ -1166,6 +1166,99 @@ Example:
...
+
+Arguments:
+
+.. csv-table::
+   :header: "Argument", "Required", "Type", "Description"
+   :align: left
+   :widths: 5,5,5,15
+
+   `timeout`, Yes, `string`, wait timeout
+   `conditions`, Yes, `map[string]interface{}`, list of wait conditions that match ``anyOf`` or ``allOf`` the conditions
```
Comment (PrasadG193) [inline docs/functions.rst:1237]: We can make it more readable and easy to consume, something like

```
conditions -> one of allOf or anyOf 
allOf/anyOf -> list of Condition ([]Conditio)

Condition:
args:
  condition: Go template condition to return tre or false
  objectReference: 
```

Let's also add a note about if someone wants to refer object key value in condition go template, they can fetch it using jsonpath syntax with prefix `$`

File: docs/functions.rst [hunk 6]
```diff
@@ -1166,6 +1166,111 @@ Example:
...
+  :linenos:
+
+  condition: "Go template condition that returns true or false"
+  objectReference:
+    apiVersion: "Kubernetes resource API version"
+    resource: "Type of resource to wait for"
+    name: "Name of the resource"
+
+.. note::
+    We can refer object key-value in Go template condition with the help of a ``$`` prefix JSON-path syntax.
```
Comment (pavannd1) [inline docs/functions.rst:1249]:
```suggestion
    We can refer to the object key-value in Go template condition with the help of a ``$`` prefix JSON-path syntax.
```

File: docs/functions.rst [hunk 7]
```diff
@@ -1166,6 +1166,111 @@ Example:
...
+                - containerPort: 80
+                  name: http
+                  protocol: TCP
+
+
+Wait
+----
+
+This function is used to wait on a Kubernetes resource
+until desired state is reached.
```
Comment (pavannd1) [inline docs/functions.rst:1225]:
```suggestion
until a desired state is reached.
```

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**